### PR TITLE
Make Json reader default, users will have to use the dataset reader for any speedups

### PR DESCRIPTION
**Pull request by avnishn**
Re our conversation about supporting JSON compatibility that the dataset reader fails to do so in some cases including:

- sample batches that are really large but are on 1 line in a json file
- inconsistent samplebatch schema in offline file. 

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by gjoliver — APPROVED** [review]
nice!

File: rllib/evaluation/worker_set.py [hunk 1]
```diff
@@ -108,53 +105,6 @@ def __init__(
             self._local_worker = None
             if num_workers == 0:
                 local_worker = True
-            if (
```
Comment (kouroshHakha) [inline rllib/evaluation/worker_set.py:111]: Do we not need this logic for backward compatibility and throwing informative errors?

Comment (avnishn) [reply rllib/evaluation/worker_set.py:111]: no this already exists in the json reader so it should be fine/further down in this file actually.

**Comment by avnishn** [comment]
looked into the failing tests looks like theyre just flakey. 

This is probably because I can't seed the dataset reader cc @clarkzinzow

**Comment by avnishn** [comment]
I'll probably just rerun the test suite tomorrow and they'll probably pass.

**Pull request closed without merging**